src/App.jsx

```diff
@@ -14,33 +17,105 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_FIELD_ALPHABETICALLY = 'alphabetically';
+const SORT_FIELD_LENGTH = 'length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+const getPreparedGoods = (
+  goods,
+  { sortField, reverseOption, setReverseOption },
+) => {
+  const preparedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  const reverseGoods = () => {
+    setReverseOption(optionList => !optionList);
+  };
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (sortField) {
+    switch (sortField) {
+      case SORT_FIELD_ALPHABETICALLY:
+        preparedGoods.sort();
+        break;
+      case SORT_FIELD_LENGTH:
+        preparedGoods.sort((good1, good2) => good1.length - good2.length);
+        break;
+      default:
+        return 0;
+    }
+  }
+
+  if (reverseOption) {
+    preparedGoods.reverse();
+  }
+
+  return { preparedGoods, reverseGoods };
+};
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverseOption, setReverseOption] = useState(false);
+  const sortedGoods = getPreparedGoods(goodsFromServer, {
+    sortField,
+    reverseOption,
+    setReverseOption,
+  });
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const reset = () => {
+    setReverseOption(false);
+    setSortField('');
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          onClick={() => setSortField(SORT_FIELD_ALPHABETICALLY)}
+          className={cn(
+            { 'is-light': sortField !== SORT_FIELD_ALPHABETICALLY },
+            'button is-info',
+          )}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          onClick={() => setSortField(SORT_FIELD_LENGTH)}
+          className={cn(
+            { 'is-light': sortField !== SORT_FIELD_LENGTH },
+            'button is-success',
+          )}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          onClick={() => sortedGoods.reverseGoods()}
+          className={cn({ 'is-light': !reverseOption }, 'button is-warning')}
+        >
+          Reverse
+        </button>
+
+        {!goodsFromServer[0]
+          .toString()
+          .includes(sortedGoods.preparedGoods[0].toString()) && (
```

Comment:
```suggestion
const isChanged = sortField ||  reverseOption;
----------------------------------------------
        {isChanged && (
```

Reply:
You can create a utility variable to check if there are some changes in the goods order